File: BJ.WEB/ClientApp/src/app/shared/models/game-views/get-unfinished-game-response-view.ts
import { SuitTypeView } from '../enums-views/suit-type-view';
import { RankTypeView } from '../enums-views/rank-type-view';

export class GetUnfinishedGameResponseView
{
    public user: UserGetUnfinishedGameResponseView;
    public bots: Array<BotGetUnfinishedGameResponseViewItem>;
    public winner: string;
/**
 *
 */
    constructor() {
      this.bots = []; 
    } 
  }
  
  export class UserGetUnfinishedGameResponseView
  {
    public name:string;
    public points: number;
    public cards: Array<StepUserGetUnfinishedGameResponseViewItem>;

    constructor() {
      this.cards = []; 
    } 

  }

  export class StepUserGetUnfinishedGameResponseViewItem
  {
    public suit: SuitTypeView;
    public rank: RankTypeView;
  }

  export class BotGetUnfinishedGameResponseViewItem
  {
    public name:string;
    public points: number;
    public cards: Array<StepBotGetUnfinishedGameResponseViewItem>;
    
    constructor() {
      this.cards = []; 
    } 
  }

  export class StepBotGetUnfinishedGameResponseViewItem
  {
    public suit: SuitTypeView;
    public rank: RankTypeView;
  }

File: BJ.ViewModels/GameViews/GetUnfinishedGameResponseView.cs
using BJ.ViewModels.EnumsViews;
using System.Collections.Generic;

namespace BJ.ViewModels.GameViews
{
    public class GetUnfinishedGameResponseView
    {

        public UserGetUnfinishedGameResponseView User { get; set; }
        public List<BotGetUnfinishedGameResponseViewItem> Bots { get; set; }
        public string Winner { get; set; }

        public GetUnfinishedGameResponseView()
        {
            Bots = new List<BotGetUnfinishedGameResponseViewItem>();
        }

    }

    public class UserGetUnfinishedGameResponseView
    {
        public string Name { get; set; }
        public int Points { get; set; }
        public List<StepUserGetUnfinishedGameResponseViewItem> Cards { get; set; }

        public UserGetUnfinishedGameResponseView()
        {
            Cards = new List<StepUserGetUnfinishedGameResponseViewItem>();
        }
    }

    public class StepUserGetUnfinishedGameResponseViewItem
    {
        public SuitTypeView Suit { get; set; }
        public RankTypeView Rank { get; set; }

    }

    public class BotGetUnfinishedGameResponseViewItem
    {
        public string Name { get; set; }
        public int Points { get; set; }
        public List<StepBotGetUnfinishedGameResponseViewItem> Cards { get; set; }

        public BotGetUnfinishedGameResponseViewItem()
        {
            Cards = new List<StepBotGetUnfinishedGameResponseViewItem>();
        }

    }

    public class StepBotGetUnfinishedGameResponseViewItem
    {
        public SuitTypeView Suit { get; set; }
        public RankTypeView Rank { get; set; }
    }


    
}
